src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -218,6 +218,32 @@ def __eq__(self, other) -> bool:
     def __hash__(self) -> int:
         return hash(hash(self.rotation_amount) + hash(tuple(self.ops_list)))
 
+    def to_basic_form_approximation(self) -> List["PauliRotation"]:
+        """Get an approximation in terms of pi/2, pi/4 and pi/8"""
+        # TODO use SK
+        axis = list(filter(lambda op: op != PauliOperator.I, self.ops_list))
+        if len(axis) != 1:
+            raise Exception("Can only approximate single qubit rotations")
+
+        if axis == PauliOperator.X:
+            raise NotImplementedError(f"decompose X rotation by {self.rotation_amount}")
+        elif axis == PauliOperator.Z:
+            raise NotImplementedError(f"decompose Z rotation by {self.rotation_amount}")
```

Comment:
@isolatedinformation this is where the code to break down arbitrary rotations will go. We could have pre built tables for Common pi/2^k Z rotations